packages/path_provider/lib/path_provider.dart

```diff
@@ -21,29 +22,46 @@ const MethodChannel _channel =
 ///
 /// On Android, this uses the `getCacheDir` API on the context.
 Future<Directory> getTemporaryDirectory() async {
-  // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-  // https://github.com/flutter/flutter/issues/26431
-  // ignore: strong_mode_implicit_dynamic_method
-  final String path = await _channel.invokeMethod('getTemporaryDirectory');
+  final String path =
+      await _channel.invokeMethod<String>('getTemporaryDirectory');
+  if (path == null) {
+    return null;
+  }
+  return Directory(path);
+}
+
+/// Path to a directory where the application may place application support
+/// files.
+///
+/// Use this for files you don’t want exposed to the user. Your app should not
+/// use this directory for user data files.
+///
+/// On iOS, this uses the `NSApplicationSupportDirectory` API.
+/// This directory is not guaranteed to exist and you can call `createSync` to
+/// create it if it doesn't exist.
+///
+/// On Android, this uses the `getDataDirectory` API on the context.
+Future<Directory> getApplicationSupportDirectory() async {
```

Comment:
A couple things:

* I think it's worth mentioning how to check that this exists too in the DartDoc if we're expecting users to check for it every time.
* If this is getting translated automatically to some other Android concept when it's called there, not sure how I feel about the iOS specific name. I'd almost rather it non-op or throw on Android if we're trying to map it to a specific iOS folder directly. WDYT?

Reply:
I'm fine with having it throw on Android for now.